static/src/javascripts/lib/url.spec.js

```diff
@@ -0,0 +1,159 @@
+// @flow
+
+import url from './url';
+
+jest.mock('lib/detect', () => ({
+    hasPushStateSupport() {
+        return true;
+    },
+}));
+
+describe('url', () => {
+    test('getUrlVars() - should get url vars', () => {
+        const origWindowSearch = window.location.search;
+        const QUERIES = [
+            ['foo', { foo: true }],
+            ['foo=bar', { foo: 'bar' }],
+            ['foo=bar&boo=far', { foo: 'bar', boo: 'far' }],
+            ['foo=bar&boo=far&', { foo: 'bar', boo: 'far' }],
+            ['foo=bar&boo', { foo: 'bar', boo: true }],
+            ['', {}],
+        ];
+
+        // pass in the query
+        QUERIES.forEach(dataProvider => {
+            const [query, expected] = dataProvider;
```

Comment:
We could destructure the tuple in the function arguments and negate the need for the cryptic `dataProvider` argument